### PR TITLE
feat(docs-infra): change navigation in resources page

### LINKED ISSUE #33526: Difficulty to explore the resource page @ https://angular.io/resources
# 📚 Docs or angular.io bug report

### Description

 **EXPLORE ANGULAR RESOURCES** @ https://angular.io/resources needs to be sturctured to be able to navigate to all resources with improved user experience. A lone scroll bar in this page will not help the reader a great deal in exploring the resources.

## 🔬 Minimal Reproduction
Go to https://angular.io/resources
### What's the affected URL?**
 https://angular.io/...

### Expected vs Actual Behavior**
User should be able to nagivate the various aspects of the angular resources with ease. The current page is unable to provide a glimpse of all the resources at one place as well.

## 📷Screenshot

image

## 🔥 Exception or Error
<pre><code>

</code></pre>

### PR DESCRIPTION
**Pull request by ajitsinghkaler**
https://angular.io/resources needs to be structured to be able to navigate to all resources with improved user experience. A lone scroll bar in this page will not help the reader a great deal in exploring the resources
Fixes #33526

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Earlier there was no navigation available on the resources page 

Issue Number: 33526

## What is the new behavior?
Added navigation to the resources page

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by ajitsinghkaler** [comment]
@gkalpak can you please look into this.

File: aio/src/app/custom-elements/resource/resource-list.component.ts [hunk 1]
```diff
@@ -28,11 +29,19 @@ export class ResourceListComponent implements OnInit {
...
 
   @HostListener('window:scroll', ['$event.target'])
   onScroll(target: any) {
     this.scrollPos = target ? target.scrollTop || target.body.scrollTop ||  0 : 0;
   }
+
+  selectCategory(id: string) {
+    this.selectedCategory =
+      this.categories.find(category => category.id.toLowerCase() === id) || this.categories[0];
+  }
```
Comment (gkalpak) [inline aio/src/app/custom-elements/resource/resource-list.component.ts]: Can you please make this component similar to the [ContributorListComponent][1] (i.e. read/write from/to the URL query string, ignore case of `id`, etc.).

[1]: https://github.com/angular/angular/blob/c7710675e/aio/src/app/custom-elements/contributor/contributor-list.component.ts

**Review by gkalpak — CHANGES_REQUESTED** [review]
Thx, @ajitsinghkaler! This looks great. A couple more tweaks and should be ready to merge (as far as I am concerned).

I think it now also makes sense to center the content under the tabs (because it looks weird to have the tabs centered and the content below left-aligned:

resources-uncentered

##
Note to self:
This should be approved by @fw-docs-marketing too (once ready).

**Comment by ajitsinghkaler** [comment]
@gkalpak should I make the content below center-aligned or should I make it take 100% of the width

**Comment by gkalpak** [comment]
I would preserve the current width.

**Comment by ajitsinghkaler** [comment]
@gkalpak can yo please review this.

File: aio/src/app/custom-elements/resource/resource-list.component.html [hunk 2]
```diff
@@ -1,28 +1,28 @@
 <div class="resources-container">
-    <div class="l-flex--column">
-      <div class="showcase" *ngFor="let category of categories">
-        <header class="c-resource-header">
-          <a class="h-anchor-offset" id="{{category.id}}"></a>
-          <h2>{{category.title}}</h2>
-        </header>
+  <div class="flex-center group-buttons">
```
Comment (gkalpak) [inline aio/src/app/custom-elements/resource/resource-list.component.html]: Thx for fixing the indentation too ❤️

File: aio/src/app/custom-elements/resource/resource-list.component.ts [hunk 3]
```diff
@@ -14,10 +15,12 @@ export class ResourceListComponent implements OnInit {
...
   location: string;
   scrollPos = 0;
+  selectedCategory: Category;
 
   constructor(
     location: PlatformLocation,
-    private resourceService: ResourceService) {
+    private resourceService: ResourceService,
+    private locationService: LocationService) {
     this.location = location.pathname.replace(/^\/+/, '');
```
Comment (gkalpak) [inline aio/src/app/custom-elements/resource/resource-list.component.ts]: Not directly related to this PR, but ?I? just noticed that the `location` property (and thus the `PlatformLocation` dependency) and the `href()` method are not used. If that is the case, could you please create another commit (I would put it before the current commit) cleaning up the component by removing these unused bits? 🙏

File: aio/src/app/custom-elements/resource/resource-list.component.ts [hunk 4]
```diff
@@ -26,13 +29,24 @@ export class ResourceListComponent implements OnInit {
...
-    this.resourceService.categories.subscribe(cats => this.categories = cats);
+    this.resourceService.categories.subscribe(cats => {
+      this.categories = cats;
+      this.selectCategory(category);
+    });
   }
 
   @HostListener('window:scroll', ['$event.target'])
   onScroll(target: any) {
     this.scrollPos = target ? target.scrollTop || target.body.scrollTop ||  0 : 0;
```
Comment (gkalpak) [inline aio/src/app/custom-elements/resource/resource-list.component.ts]: Not directly related to this PR, but the `scrollPos` property and the `onScroll()` method don't seem to be used. Could you include them in the cleanup commit mentioned in a previous comment?

**Review by gkalpak — CHANGES_REQUESTED** [review]
Thx, @ajitsinghkaler :+1: Yes, please, add some tests for the new functionality (i.e. toggling between categories, showing entries from a specific category only and interacting with the URL query).

I have left some comments to do some clean-up (which is not directly related to this PR, but would be awesome if you could do (on a separate commit)).

Other than that, this lgtm (pending the addition of tests) 👍

**Comment by ajitsinghkaler** [comment]
@gkalpak I've added the cleanup please have a look

File: aio/src/app/custom-elements/resource/resource-list.component.ts (hunk 3 above)
Comment (gkalpak) [reply aio/src/app/custom-elements/resource/resource-list.component.ts]: Can you please remove the `location` property (which doesn't seem to be used)?

**Review by gkalpak — CHANGES_REQUESTED** [review]
The tests are failing, because the injector that is set up for the tests must be updated to provide the necessary dependencies [here][1]. I.e. remove `PlatformLocation` and provide a mock `Location` service (you can use [this one][2]).

<sub>Don't forget to add tests for the new functionality (as mentioned above).</sub>

[1]: https://github.com/angular/angular/blob/91ea8d6f3bb4bb9a2ac77839024f34ca52e03b39/aio/src/app/custom-elements/resource/resource-list.component.spec.ts#L21
[2]: https://github.com/angular/angular/blob/91ea8d6f3bb4bb9a2ac77839024f34ca52e03b39/aio/src/testing/location.service.ts

**Comment by ajitsinghkaler** [comment]
@gkalpak added the tests please have alook

**Review by gkalpak — CHANGES_REQUESTED** [review]
Almost there :smiley:
I left some minor clean-up suggestions for the tests. Other than that, you just need to get CI green and this is good to go.

**Comment by ajitsinghkaler** [comment]
@gkalpak I've made all the changes. @sjtrimble can you please review my pull request

**Review by gkalpak — APPROVED** [review]

**Review by sjtrimble — APPROVED** [review]
Nice!

**Review by gkalpak — CHANGES_REQUESTED** [review]
I just noticed that the commit messages need some clean up (to follow our [giudelines][1]).
Can you please squash the last two commits into one (so that the PR has overall two commits) and make sure there is a blank line above `Fixes ...` in the second commit message?

[1]: https://github.com/angular/angular/blob/c1cf46c5c/CONTRIBUTING.md#commit

**Comment by ajitsinghkaler** [comment]
@gkalpak made the changes please have a look

**Review by gkalpak — APPROVED** [review]

**Comment by ajitsinghkaler** [comment]
@gkalpak can you please add the merge label

**Pull request closed without merging**